diff --git CMakeLists.txt CMakeLists.txt
index 673f40b..3b4c490 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -70,6 +70,7 @@ option(MINIAUDIO_NO_NEON                       "Disable NEON optimizations"
 option(MINIAUDIO_NO_RUNTIME_LINKING            "Disable runtime linking"             OFF)
 option(MINIAUDIO_USE_STDINT                    "Use <stdint.h> for sized types"      OFF)
 option(MINIAUDIO_DEBUG_OUTPUT                  "Enable stdout debug output"          OFF)
+option(MINIAUDIO_DISABLE_INSTALL               "Disable installation"                OFF)
 
 
 include(GNUInstallDirs)
@@ -508,7 +509,9 @@ add_library(miniaudio
 )
 
 list(APPEND LIBS_TO_INSTALL miniaudio)
-install(FILES miniaudio.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/miniaudio)
+if(NOT MINIAUDIO_DISABLE_INSTALL)
+    install(FILES miniaudio.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/miniaudio)
+endif()
 
 target_include_directories(miniaudio PUBLIC  ${CMAKE_CURRENT_SOURCE_DIR})
 target_compile_options    (miniaudio PRIVATE ${COMPILE_OPTIONS})
@@ -534,7 +537,9 @@ if(HAS_LIBVORBIS)
     )
 
     list(APPEND LIBS_TO_INSTALL miniaudio_libvorbis)
-    install(FILES extras/decoders/libvorbis/miniaudio_libvorbis.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/miniaudio/extras/decoders/libvorbis)
+    if(NOT MINIAUDIO_DISABLE_INSTALL)
+        install(FILES extras/decoders/libvorbis/miniaudio_libvorbis.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/miniaudio/extras/decoders/libvorbis)
+    endif()
 
     target_compile_options    (miniaudio_libvorbis PRIVATE ${COMPILE_OPTIONS})
     target_compile_definitions(miniaudio_libvorbis PRIVATE ${COMPILE_DEFINES})
@@ -562,7 +567,9 @@ if(HAS_LIBOPUS)
 
 
     list(APPEND LIBS_TO_INSTALL miniaudio_libopus)
-    install(FILES extras/decoders/libopus/miniaudio_libopus.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/miniaudio/extras/decoders/libopus)
+    if(NOT MINIAUDIO_DISABLE_INSTALL)
+        install(FILES extras/decoders/libopus/miniaudio_libopus.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/miniaudio/extras/decoders/libopus)
+    endif()
 
     target_compile_options    (miniaudio_libopus PRIVATE ${COMPILE_OPTIONS})
     target_compile_definitions(miniaudio_libopus PRIVATE ${COMPILE_DEFINES})
@@ -581,7 +588,9 @@ if (NOT MINIAUDIO_NO_EXTRA_NODES)
 
         list(APPEND libs miniaudio_${name}_node)
         set(LIBS_TO_INSTALL "${libs}" PARENT_SCOPE) # without PARENT_SCOPE, any changes are lost
-        install(FILES extras/nodes/ma_${name}_node/ma_${name}_node.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/miniaudio/extras/nodes/ma_${name}_node)
+        if(NOT MINIAUDIO_DISABLE_INSTALL)
+            install(FILES extras/nodes/ma_${name}_node/ma_${name}_node.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/miniaudio/extras/nodes/ma_${name}_node)
+        endif()
 
         target_include_directories(miniaudio_${name}_node PUBLIC  ${CMAKE_CURRENT_SOURCE_DIR}/extras/nodes/ma_${name}_node)
         target_compile_options    (miniaudio_${name}_node PRIVATE ${COMPILE_OPTIONS})
@@ -856,11 +865,13 @@ string(JOIN " " MINIAUDIO_PC_CFLAGS ${COMPILE_DEFINES})
 
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/miniaudio.pc.in" "${CMAKE_CURRENT_BINARY_DIR}/miniaudio.pc" @ONLY)
 
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/miniaudio.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+if(NOT MINIAUDIO_DISABLE_INSTALL)
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/miniaudio.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 
-message(STATUS "Library list: ${LIBS_TO_INSTALL}")
-install(TARGETS ${LIBS_TO_INSTALL}
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-)
+    message(STATUS "Library list: ${LIBS_TO_INSTALL}")
+    install(TARGETS ${LIBS_TO_INSTALL}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    )
+endif()
diff --git miniaudio.h miniaudio.h
index 1e4e2c6..c1e3588 100644
--- miniaudio.h
+++ miniaudio.h
@@ -61653,7 +61653,7 @@ static ma_result ma_default_vfs_info(ma_vfs* pVFS, ma_vfs_file file, ma_file_inf
 
     if (result == MA_NOT_IMPLEMENTED) {
         /* Not implemented. Fall back to seek/tell/seek. */
-        ma_result result;
+        // ma_result result;
         ma_int64 cursor;
         ma_int64 sizeInBytes;
         
@@ -71356,29 +71356,21 @@ MA_API ma_result ma_resource_manager_data_buffer_read_pcm_frames(ma_resource_man
         isDecodedBufferBusy = (ma_resource_manager_data_buffer_node_result(pDataBuffer->pNode) == MA_BUSY);
 
         if (ma_resource_manager_data_buffer_get_available_frames(pDataBuffer, &availableFrames) == MA_SUCCESS) {
-            /* Don't try reading more than the available frame count if the data buffer node is still loading. */
-            if (isDecodedBufferBusy) {
-                if (frameCount > availableFrames) {
-                    frameCount = availableFrames;
+            if (frameCount > availableFrames) {
+                frameCount = availableFrames;
 
-                    /*
-                    If there's no frames available we want to set the status to MA_AT_END. The logic below
-                    will check if the node is busy, and if so, change it to MA_BUSY. The reason we do this
-                    is because we don't want to call `ma_data_source_read_pcm_frames()` if the frame count
-                    is 0 because that'll result in a situation where it's possible MA_AT_END won't get
-                    returned.
-                    */
-                    if (frameCount == 0) {
-                        result = MA_AT_END;
-                    }
-                } else {
-                    isDecodedBufferBusy = MA_FALSE; /* We have enough frames available in the buffer to avoid a MA_BUSY status. */
-                }
-            } else {
                 /*
-                Getting here means the buffer has been fully loaded. We can just pass the frame count straight
-                into ma_data_source_read_pcm_frames() below and let ma_data_source handle it.
+                If there's no frames available we want to set the status to MA_AT_END. The logic below
+                will check if the node is busy, and if so, change it to MA_BUSY. The reason we do this
+                is because we don't want to call `ma_data_source_read_pcm_frames()` if the frame count
+                is 0 because that'll result in a situation where it's possible MA_AT_END won't get
+                returned.
                 */
+                if (frameCount == 0) {
+                    result = MA_AT_END;
+                }
+            } else {
+                isDecodedBufferBusy = MA_FALSE; /* We have enough frames available in the buffer to avoid a MA_BUSY status. */
             }
         }
     }
