diff --git src/json-patch.cpp src/json-patch.cpp
--- src/json-patch.cpp
+++ src/json-patch.cpp
@@ -4,7 +4,7 @@
 
 namespace
 {
-
+using namespace nlohmann::literals;
 // originally from http://jsonpatch.com/, http://json.schemastore.org/json-patch
 // with fixes
 const nlohmann::json patch_schema = R"patch({
diff --git src/json-validator.cpp src/json-validator.cpp
--- src/json-validator.cpp
+++ src/json-validator.cpp
@@ -907,7 +907,7 @@ class numeric : public schema
 
 	void validate(const json::json_pointer &ptr, const json &instance, json_patch &, error_handler &e) const override
 	{
-		T value = instance; // conversion of json to value_type
+		T value = instance.get<T>(); // conversion of json to value_type
 
 		std::ostringstream oss;
 
