diff --git XoshiroCpp.hpp XoshiroCpp.hpp
index 585c962..1ecfdfb 100644
--- XoshiroCpp.hpp
+++ XoshiroCpp.hpp
@@ -70,7 +70,7 @@ namespace XoshiroCpp
 
 		constexpr result_type operator()() noexcept;
 
-		template <size_t N>
+		template <std::size_t N>
 		[[nodiscard]]
 		constexpr std::array<std::uint64_t, N> generateSeedSequence() noexcept;
 
@@ -689,7 +689,7 @@ namespace XoshiroCpp
 		return z ^ (z >> 31);
 	}
 
-	template <size_t N>
+	template <std::size_t N>
 	inline constexpr std::array<std::uint64_t, N> SplitMix64::generateSeedSequence() noexcept
 	{
 		std::array<std::uint64_t, N> seeds = {};
